test/built-ins/Promise/prototype/finally/rejected-observable-then-calls.js

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2017 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+author: Jordan Harband
+description: finally observably calls .then
+esid: sec-promise.prototype.finally
+features: [Promise.prototype.finally]
+flags: [async]
+---*/
+
+var initialThenCount = 0;
+var noReason = {};
+var no = Promise.reject(noReason);
+no.then = function () {
+  initialThenCount += 1;
+  return Promise.prototype.then.apply(this, arguments);
+};
+
+var onFinallyThenCount = 0;
+var yesValue = {};
+var yes = Promise.resolve(yesValue);
+yes.then = function () {
+  onFinallyThenCount += 1;
+  return Promise.prototype.then.apply(this, arguments);
+};
+
+var finallyCalled = false;
+var catchCalled = false;
+
+no.catch(function (e) {
+  assert.sameValue(e, noReason);
+  throw e;
```

Comment:
Ohh, I see. You're right. 
So it will be enough to check if `finally` returns `Promise`.  Not sure that I can find such test, or I missed something.
If there is no such test we can use `Symbol.toStringTag`:
```
const p = Promise.resolve(1);
assert.sameValue(p.finally()[Symbol.toStringTag], 'Promise');
```